classes/webservice/WebserviceOutputBuilder.php

```diff
@@ -278,25 +285,28 @@ public function setStatus($num)
      */
     public function getErrors($errors)
     {
-        $str_output = '';
-        if (!empty($errors)) {
-            if (isset($this->objectRender)) {
-                $str_output = $this->objectRender->renderErrorsHeader();
-                foreach ($errors as $error) {
-                    if (is_array($error)) {
-                        $str_output .= $this->objectRender->renderErrors($error[1], $error[0]);
-                    } else {
-                        $str_output .= $this->objectRender->renderErrors($error);
-                    }
+        if (empty($errors)) {
+            return '';
+        }
+
+        if (!isset($this->objectRender)) {
+            return '<pre>' . print_r($errors, true) . '</pre>';
+        }
+
+        $this->output = ApiNode::list('errors');
```

Comment:
Do we really need to use Singleton?

Reply:
Its not actually singleton pattern, just a shortcut to utilize multiple constructors. You could replace that with Factory or with simple constructor with fluent setters.
Factory would need another class.
Constructor with fluent setters gets more error-prone for future development, in my opinion.